### PR TITLE
feat(primitives): remove `mod ethers_compat`

### LINKED ISSUE #4738: Remove ethers conversions (`mod ethers_compat` etc) in `reth-primitives`
### :warning: Base on #4737 (`alloy`) :warning:

Goal: get rid of `ethers-core` in `reth-primitives` as much as possible

### PR DESCRIPTION
**Pull request by tcoratger**
Should close #4738.

### CODE REVIEW

**Comment by tcoratger** [comment]
This is just a draft here, I'm not that familiar with `alloy` but I think that we need to change `GethInstance` in:

https://github.com/paradigmxyz/reth/blob/272f06216a65750043e59b6064b5b31a3f402b3d/crates/net/network/tests/it/clique/geth.rs#L36-L41

in order to remove the `ethers_core` dependency here. Maybe this is is part of https://github.com/paradigmxyz/reth/pull/4737 or to be included here after the merge?

**Comment by DaniPopes** [comment]
Ah they're used for tests. This is fine for now, you can make the modules `#[cfg(feature = "test-utils")]`

**Comment by tcoratger** [comment]
> Ah they're used for tests. This is fine for now, you can make the modules `#[cfg(feature = "test-utils")]`

@DaniPopes Done.

File: crates/net/network/Cargo.toml [hunk 1]
```diff
@@ -19,7 +19,7 @@ normal = [
 [dependencies]
 # reth
 reth-interfaces.workspace = true
-reth-primitives.workspace = true
+reth-primitives = { workspace = true, features = ["test-utils"] }
```
Comment (DaniPopes) [inline crates/net/network/Cargo.toml:22]: This should be enabled just for dev-dependencies

**Review by DaniPopes — CHANGES_REQUESTED** [review]

File: crates/net/network/Cargo.toml (hunk 1 above)
Comment (tcoratger) [reply crates/net/network/Cargo.toml]: @DaniPopes Done

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**